### ISSUE TITLE
Build parser for the Wage and Income Tax Transcript

### ISSUE THREAD

**Issue by zakpatterson** [ustaxes/UsTaxes#1008]
Taxpayers that receive any statement reportable to the IRS with their SSN should have a "Wage and Income Tax Transcript" [Available here](https://www.irs.gov/individuals/get-transcript). 

This PDF, when downloaded, is copy pasteable, not rasterized, so it should be parseable.

**Comment by gtoubassi** [comment]
I am super excited about this possibility but I think the problem is that this info is not available until too late, possibly only available once the return is filed.  For example as of two days ago my 2021 wage/tax transcript is still empty (even though of course I've received various 1099's and thus so has the IRS!).  In fact it says something like "No return on file" so it may be a thing that they only publish it late which is super lame.  Don't take my word for it you should check yours and ask a few others in the community to check to make sure.

**Comment by tmm1** [comment]
It does appear as though the transcript itself is based on your return:

>The method you used to file your tax return (e-file or paper) and whether you had a balance due affects  your [current year transcript availability](https://www.irs.gov/individuals/transcript-availability).

According to that doc, the earliest you can receive a transcript is 2-3 weeks *after* you file your return.

**Comment by gtoubassi** [comment]
Nice find on the docs.  I hadn't seen that.  Just changing that policy would really unlock some possibilities.  Its possible there is some questionable logic like "if we tell taxpayers what we know then they may decide not to tell us about stuff we don't know, so better to keep them guessing."

**Comment by tmm1** [comment]
Just to confirm, I requested my 2021 Wage and Income Transcript and it says "No record of return filed."

Seems to me like the data is not present in this system until the return is filed and processed.

**Comment by gtoubassi** [comment]
Yes thats what I get.  But the IRS does have this correct?  By law those 1099's have already hit their systems.  Maybe it hasn't propagated to whatever system actually can render the transcript but they have the data somewhere, no?

**Comment by zakpatterson** [comment]
This is nuts. I know this used to work because there was always a form I was missing and I would use this to get the details I needed. 

Anyway if this is really how it is then maybe we all should just file a blank 1040 in February and then amend before the deadline.

**Comment by thegrims** [comment]
Plaid has a payroll API that might be useful for gathering W2 information
https://plaid.com/docs/api/products/income/#creditpayroll_incomeget

**Comment by zakpatterson** [comment]
Our whole mission is to protect user privacy and user data. I don't think we can interop with plaid.

**Comment by gtoubassi** [comment]
@zakpatterson that's really interesting you have been able to download the transcript before the filing date!  Have you tried the "head fake" to see if you can unlock the data?  I'm somehow not excited about filing a blank or temporary return :).  But that's super good to hear that maybe this is not a deep policy thing but a bug or something that has a workaround.

**Comment by zakpatterson** [comment]
@gtoubassi This was a few years ago at least. Also I was definitely under extension, so it's possible that this transcript opens up if a user files for an extension. 

By no means suggesting people wrongly file a zero income return ha..

**Comment by tmm1** [comment]
I plan to file an extension this year so I can verify that theory.

**Comment by gtoubassi** [comment]
Actually I'm doing an extension as well so will check as well!

**Comment by zakpatterson** [comment]
![image](https://user-images.githubusercontent.com/1987109/161674655-0a7b4e6f-b59d-447a-899c-b22f3fb2b880.png)

The IRS page shows this for the wage and income transcript. I don't know why the PDF, if downloaded, displays "No record of return filed". Maybe that's actually referring to forms that reporters are required to file that match to taxpayer's ssn.

**Comment by tmm1** [comment]
Checked my transcript again today and it's still empty.

Also saw this today, only tangentially related but interesting: https://twitter.com/goodmanl/status/1521097438127435777?s=21&t=tN1WKGxuKueyHK60GTiOPA

**Comment by gtoubassi** [comment]
Yes I checked as well a few days ago and saw the same.  Starting to assume it just pops in once you file (perhaps with some propagation delay), but we will see.  Thanks for the tweet reference!  Interesting.

**Comment by tmm1** [comment]
My transcript now contains W2 and 1099 data even though I have not filed my taxes.

**Comment by gtoubassi** [comment]
Same here!  I checked just a few weeks ago (well after the filing deadline) and it was still empty.
